### ISSUE TITLE
relational_tucker3.relation_embedder.base_embedder.dim = -1 causes error

### ISSUE THREAD

**Issue by tsafavi** [uma-pi1/kge#118]
When I try to run an RT3 training job without explicitly setting `relational_tucker3.relation_embedder.base_embedder.dim`, I get a PyTorch error about creating a Tensor with a negative dimension. This is because the default config file (kge/model/relational_tucker3.yaml) sets `relational_tucker3.relation_embedder.base_embedder.dim = -1`, which doesn't appear to be updated anywhere when the embeddings are initialized. Is the -1 supposed to be referring to `relational_tucker3.relation_embedder.dim` (which is automatically set to the entity embedding dim ** 2) perhaps?  

Config file to replicate error: 
```
job.type: train
dataset.name: toy
model: relational_tucker3

relational_tucker3:
  entity_embedder:
    dim: 50
    dropout: 0.02
    regularize: 'lp'
    regularize_args:
      weight: 4e-12

  relation_embedder:
    dropout: 0.3
    regularize: 'lp'
    regularize_args:
      weight: 6e-16

train:
  max_epochs: 50
  optimizer: Adagrad
  optimizer_args:
    lr: 0.1
```

**Comment by samuelbroscheit** [comment]
`relational_tucker3.relation_embedder.base_embedder.dim` has to be set explicitly 

<img src="https://raw.githubusercontent.com/ibalazevic/TuckER/master/tucker.png"/>

d_r and d_e are not depending on each other like in other KGE models.

**Comment by tsafavi** [comment]
Is it worth leaving a note in the documentation that it needs to be set explicitly? Other default configs use -1 to mean that a value will get set automatically, but here that is not the case so it could be confusing for users.

**Comment by rgemulla** [comment]
To clarify: 
- `relational_tucker3.relation_embedder.base_embedder.dim` (d_r) refers to the dimensions used to represent a relation and has to be specified manually. 
- `relational_tucker3.relation_embedder.dim` refers to the required embedding size for a relation. For Tucker3, this is d_e^2.
- What the RT3 embedder does by default is to project the $d_r$-sized base embedding to size $d_e^2$.

**Comment by rgemulla** [comment]
Yes, it is. I'll reopen this until the docs are updated.

**Comment by tsafavi** [comment]
The behavior also seems inconsistent with how entity embeddings are initialized. If I understand correctly, entity embeddings will always be initialized with a default size if not specified by the user. Why not also initialize the RT3 relation embedding dimension with a default value if not explicitly set?

**Comment by samuelbroscheit** [comment]
Makes sense to expand the documentation!

I would vote against a default size, because it could lead to users assuming that there is a reasonable default size that we suggest to use.

**Comment by samuelbroscheit** [comment]
> Edit: The default value actually seems to be 100, but this is overwritten by the -1 in relational_tucker3.yaml.

Hm, I can run https://github.com/uma-pi1/kge/blob/master/examples/toy-rt3-train.yaml without problems. How does this manifest itself?

**Comment by tsafavi** [comment]
The config snippet I pasted is not exactly the file you linked - lines 18 through 23 have been deleted in the snippet (everything indented under `base_embedder`). With this snippet, I get the error: `RuntimeError: Trying to create tensor with negative dimension -1: [112, -1]`

**Comment by samuelbroscheit** [comment]
Ah ok, I misread what you wrote. The error is expected when base_embedder.dim is not set.

**Comment by rgemulla** [comment]
Addressed in #121
